src/lib.rs

```diff
@@ -196,14 +196,66 @@ macro_rules! custom_error {
                 match self {$(
                     $errtype::$field $( { $( $attr_name ),* } )* => {
                         $(write!(formatter, "{}", ($($msg_fun)*) )?;)*
-                        $crate::display_message!(formatter, $($($attr_name),*),* | $($msg)*);
+                        $crate::display_message_for_enum!(formatter, $($($attr_name),*),* | $($msg)*);
                         Ok(())
                     }
                 ),*}
             }
         }
         }}
     };
+    (
+        $( ($prefix:tt) )* // `pub` marker
+        $errtype:ident // Name of the error type to generate
+        $( < $(
+            $type_param:tt // Optional type parameters for generic error types
+            ),*
+        > )*
+        { $(
+            $field_name:ident // Name of an attribute of the error variant
+            :
+            $($field_type:ident)::* // type of the attribute
+            $(< $($field_type_param:tt),* >)* // Generic (lifetime & type) parameters for the attribute's type
+        ),* }
+        =
+        $( @{ $($msg_fun:tt)* } )*
```

Comment:
If advanced custom error messages are supported, could you please add a test for them ?